atlasaction/action.go

```diff
@@ -24,6 +32,38 @@ func MigrateApply(ctx context.Context, client *atlasexec.Client, act *githubacti
 		TxMode:          act.GetInput("tx-mode"),  // Hidden param.
 		BaselineVersion: act.GetInput("baseline"), // Hidden param.
 	}
+	// Cloud-based migration directory.
+	if act.GetInput("dir-name") != "" {
+		if params.DirURL != "" {
+			return errors.New("dir and dir-name are mutually exclusive")
+		}
+		// Cloud-based migrations are currently based on creating a temporary atlas.hcl
+		// file therefore it cannot be used with a user-supplied config.
```

Comment:
Is it implemented this way because of the override of `cloud-url` and `cloud-token`?
If so, I think this should be solved in the testing code, not inside the production code and causing a limitation on the users